test/scenarios/bookshop/runtime-views.test.js

```diff
@@ -0,0 +1,227 @@
+          id: 207,
+          title: 'Jane Eyre',
+          AuthorName: 'Charlotte Brontë',
+        })
+        expect(res[1]).toMatchObject({
+          id: 201,
+          title: 'Wuthering Heights',
+          AuthorName: 'Emily Brontë',
+        })
+
+        const resDeployed = await SELECT.from('views2Service.Book')
+          .columns(['id', 'title', 'AuthorName'])
+          .where('id = 201 or id = 207')
+          .orderBy('title')
+        expect(res).toEqual(resDeployed)
+      })
+    })
+
+    describe('Aggregations and Distinct Queries', () => {
+      test('distinct author selection', async () => {
+        const res = await SELECT.from('runtimeViews0Service.Book')
+          .columns([{ ref: ['author'], expand: ['*'] }])
+          .where('ID != 201')
+
+        const authors = res.map(b => b.author.name)
+        expect(authors).toEqual(expect.arrayContaining(['Charlotte Brontë', 'Edgar Allen Poe', 'Richard Carpenter']))
+
+        const resDeployed = await SELECT.from('views0Service.Book')
+          .columns([{ ref: ['author'], expand: ['*'] }])
+          .where('ID != 201')
+        expect(res).toEqual(resDeployed)
+      })
+
+      test('group by with aggregate count', async () => {
+        const res = await SELECT.from('runtimeViews2Service.Book')
+          .columns(['AuthorName', 'count(*) as books'])
+          .where('AuthorName IS NOT NULL')
+          .groupBy('AuthorName')
+
+        expect(res).toEqual(
+          expect.arrayContaining([
+            { AuthorName: 'Charlotte Brontë', books: 1 },
+            { AuthorName: 'Emily Brontë', books: 1 },
+          ]),
+        )
+
+        const resDeployed = await SELECT.from('views2Service.Book')
+          .columns(['AuthorName', 'count(*) as books'])
+          .where('AuthorName IS NOT NULL')
+          .groupBy('AuthorName')
+        expect(res).toEqual(resDeployed)
+      })
+    })
+
+    describe('Redirected views', () => {
+      test('runtime with books redirected', async () => {
+        const res = await SELECT.from('runtimeViews0Service.AuthorRedirected').columns(['ID', { expand: [{ref: ['title']}], ref: ['books'] }]).where({ ID: 101 })
+        expect(res).toHaveLength(1)
+        expect(res[0]).toMatchObject({
+          ID: 101,
+          books: [
+            { title: 'Redirected Wuthering Heights' }
+          ]
+        })
+        const resDeployed = await SELECT.from('views0Service.AuthorRedirected').columns(['ID', { expand: [{ref: ['title']}], ref: ['books'] }]).where({ ID: 101 })
+        expect(res).toEqual(resDeployed)
+      })
+    })
+  })
+
+  describe('Error Cases', () => {
+    describe('Unsupported Entity Types', () => {
+      test('Virtual entities should throw error', async () => {
+        try {
+          await SELECT.from('runtimeViewsErrorService.VirtualBookView')
+          fail('Expected request to throw an error')
+        } catch (error) {
+          expect(error.message).toMatch(/is not a runtime view/)
+        }
```

Comment:
`expect` comes with the `rejected` function to ensure that the promise throws as expected.

```suggestion
        const { VirtualBookView } = cds.entities('runtimeViewsErrorService')
        await expect(cds.ql`select from ${VirtualBookView }).rejected
          .to.eq(/is not a runtime view/) // the way to validate the error object is not very straight forward
```